app/services/topic_status_updater.rb

```diff
@@ -46,6 +46,14 @@ def change(status, opts = {})
       UserProfile.remove_featured_topic_from_all_profiles(topic)
     end
 
+    if status.visible?
+      if status.enabled?
+        topic.update_category_topic_count_by(1)
+      else
+        topic.update_category_topic_count_by(-1)
+      end
```

Comment:
Not sure it's easier to understand but you could also do

```suggestion
      topic.update_category_topic_count_by(status.enabled? ? 1 : -1)
```